### DISCUSSION TITLE
system stops charging below 26V

### DISCUSSION THREAD

**Discussion by hansgeorgk** [Louisvdw/dbus-serialbattery#278] [Q&A] [answered]
hi,

I just switched to an EasySolar 2 GX 24/3000 and new new Daly BMS 24V 280AH (8 x EVE cells). I realized that even with full sun, the system is only charging with 3-20W, when I would be expecting 200-400 Watts. I rebooted the Gx device, which sometimes solved the problem for a few minutes.
Today the battery ran empty and I connected a Generator to the EasySolar. It charged for a few minutes, then the MPPT switched off with Err#38/39 and the inverter switched to passthrough mode, not charging the batteries.

I doublechecked all settings - MPPT, BMS, utils.py
MPPT is set to 24V. 
any idea where else to check for Voltage settings?
Is there any Voltage autodetection in the dbus-serialdriver

cheers
Hans-Georg

**Comment by Louisvdw** [comment]
The voltage is set up from the cell count. (cell # x 3.45V)
Check the VRM graphs (advance tab) which will show you the voltages and charge limits over time. https://github.com/Louisvdw/dbus-serialbattery/wiki/Troubleshoot#check-your-graphs-in-advance-section-in-your-vrm

**Comment by hansgeorgk** [comment]
have a look at the graphs from earlier this week. we had nice sun all day long, but the MPPT is charging with less than 20W :-( for a long period and then picking up again. 
Do you have any idea what is going on? Any other log files or debug mode available?

<img width="352" alt="Bildschirmfoto 2022-10-30 um 05 49 38" src="https://user-images.githubusercontent.com/100286224/198863138-f480351d-12a8-44e2-8bfb-06e6521e4b21.png">
<img width="358" alt="Bildschirmfoto 2022-10-30 um 05 49 55" src="https://user-images.githubusercontent.com/100286224/198863140-5c4d8ce9-5fe6-4c83-947a-5751366254be.png">

**Reply by Louisvdw** [reply, marked as answer]
That is how your system should work.

When your battery if fully charged the MPPTs will start to limit to power generation to only what your system use as there is nothing more to store in the batteries. So you need to look at the SOC and Charge limit graphs along with the PV. As you can see in the graphs below the PV ramps up to around 11am and then when the battery is fully charged the PV is reduced to only what is consumed. So when you start using energy the PV will increase PV power to supply that load. The Output power graph will match the wiggles after 11am (not shown below).

![image](https://user-images.githubusercontent.com/26077665/198883471-ba972604-8d02-44f2-84f6-7b346d71379d.png)

**Comment by hansgeorgk** [comment]
look at my graphs - the batteries are not full, but still the MPPT stops charging :-(
<img width="368" alt="Bildschirmfoto 2022-10-30 um 21 23 01" src="https://user-images.githubusercontent.com/100286224/198900022-66b1f466-59fe-490a-b6cf-f6bc5fcec78d.png">
<img width="362" alt="Bildschirmfoto 2022-10-30 um 21 22 29" src="https://user-images.githubusercontent.com/100286224/198900030-a431dab9-858f-4ef0-ac3a-f78607ea909c.png">
<img width="360" alt="Bildschirmfoto 2022-10-30 um 21 21 29" src="https://user-images.githubusercontent.com/100286224/198900038-2a69c771-4383-4606-a203-afad3b99e00c.png">

As far as I can see, the question now is, why/how is the MPPT getting the wrong info and stops charging

**Reply by Louisvdw** [reply]
Can you show what settings do you have in the GX under the DVCC menu
